force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -212,10 +255,17 @@ public with sharing class ClassesObjectsMethods {
      * @return The a positive number years between the two dates, or null if either date is null.
      */
     public static Integer calculateYearDifference(Date d1, Date d2) {
-
+        Integer yearsBetween;
         // Subtract the difference in years
+        if(d1 == null || d2 == null){
+            yearsBetween = null;
+        }
+        else{
+            yearsBetween = math.abs(d1.year() - d2.year());
+        }
+
 
-        return null; // Replace null with the variable you used to store the result
+        return yearsBetween; // Replace null with the variable you used to store the result
```

Comment:
Good work. I suggest using the replace conditional with guard method I shared in Slack a while back. Each branch of conditional logic you write needs to be tested separately, and it is another place for a bug to hide. I try to be ruthless in removing conditional logic from my code. 

```
 public static Integer calculateYearDifference(Date d1, Date d2) {
        if (d1 == null || d2 == null) {
            return null;
        }
        return Math.abs(d1.year() - d2.year());
    }
 ```